internal/featuresupport/feature_support_level.go

```diff
@@ -0,0 +1,91 @@
+	models.FeatureSupportLevelIDDISKSELECTION:                      &DiscSelectionFeature{},
+	models.FeatureSupportLevelIDOVNNETWORKTYPE:                     &OvnNetworkTypeFeature{},
+	models.FeatureSupportLevelIDSDNNETWORKTYPE:                     &SdnNetworkTypeFeature{},
+	models.FeatureSupportLevelIDPLATFORMSELECTION:                  &PlatformSelectionFeature{},
+	models.FeatureSupportLevelIDSCHEDULABLEMASTERS:                 &SchedulableMastersFeature{},
+	models.FeatureSupportLevelIDAUTOASSIGNROLE:                     &AutoAssignRoleFeature{},
+	models.FeatureSupportLevelIDCUSTOMMANIFEST:                     &CustomManifestFeature{},
+	models.FeatureSupportLevelIDDISKENCRYPTION:                     &DiskEncryptionFeature{},
+	models.FeatureSupportLevelIDCLUSTERMANAGEDNETWORKINGWITHVMS:    &ClusterManagedNetworkingWithVmsFeature{},
+	models.FeatureSupportLevelIDCLUSTERMANAGEDNETWORKING:           &ClusterManagedNetworkingFeature{},
+	models.FeatureSupportLevelIDSINGLENODEEXPANSION:                &SingleNodeExpansionFeature{},
+	models.FeatureSupportLevelIDLVM:                                &LvmFeature{},
+	models.FeatureSupportLevelIDDUALSTACKNETWORKING:                &DualStackNetworkingFeature{},
+	models.FeatureSupportLevelIDNUTANIXINTEGRATION:                 &NutanixIntegrationFeature{},
+	models.FeatureSupportLevelIDDUALSTACKVIPS:                      &DualStackVipsFeature{},
+	models.FeatureSupportLevelIDUSERMANAGEDNETWORKINGWITHMULTINODE: &UserManagedNetworkingWithMultiNodeFeature{},
+}
+
+var cpuFeaturesList = map[models.ArchitectureSupportLevelID]SupportLevelArchitecture{
+	models.ArchitectureSupportLevelIDX8664ARCHITECTURE:     &X8664ArchitectureFeature{},
+	models.ArchitectureSupportLevelIDARM64ARCHITECTURE:     &Arm64ArchitectureFeature{},
+	models.ArchitectureSupportLevelIDS390XARCHITECTURE:     &S390xArchitectureFeature{},
+	models.ArchitectureSupportLevelIDPPC64LEARCHITECTURE:   &PPC64LEArchitectureFeature{},
+	models.ArchitectureSupportLevelIDMULTIARCHRELEASEIMAGE: &MultiArchReleaseImageFeature{},
+}
+
+func getFeatureSupportList(features map[models.FeatureSupportLevelID]SupportLevelFeature, filters SupportLevelFilters) models.SupportLevels {
+	featureSupportList := models.SupportLevels{}
+
+	for _, feature := range features {
+		if reflect.TypeOf(feature).Name() == "SupportLevelFeature" {
+			featureID := feature.GetId()
+			featureSupportList[string(featureID)] = feature.GetSupportLevel(filters)
+		}
+	}
+	return featureSupportList
+}
+
+func getArchitectureSupportList(features map[models.ArchitectureSupportLevelID]SupportLevelArchitecture, openshiftVersion string) models.SupportLevels {
+	featureSupportList := models.SupportLevels{}
+
+	for _, feature := range features {
+		featureID := feature.GetId()
+		featureSupportList[string(featureID)] = feature.GetSupportLevel(openshiftVersion)
+	}
+	return featureSupportList
+}
+
+func GetFeatureSupportList(filters SupportLevelFilters) models.SupportLevels {
```

Comment:
My goal was to avoid a situation where we would have to update every feature signature if we wanted to add a parameter later on.
If you examine the features object closely, you'll see that the CPU architecture is often irrelevant, so its inclusion in the code can be aesthetically unappealing.